src/Functions/DateTimeTransforms.h

```diff
@@ -1518,6 +1518,31 @@ struct ToSecondImpl
     using FactorTransform = ToStartOfMinuteImpl;
 };
 
+struct ToMillisecondImpl
+{
+    static constexpr auto name = "toMillisecond";
+
+    static UInt16 execute(Int64 t, const DateLUTImpl & time_zone)
+    {
+        return time_zone.toSecond(t)*1000;
+    }
+    static UInt16 execute(UInt32 t, const DateLUTImpl & time_zone)
+    {
+        return time_zone.toSecond(t)*1000;
+    }
+    static UInt16 execute(Int32, const DateLUTImpl &)
+    {
+        throwDateIsNotSupported(name);
+    }
+    static UInt16 execute(UInt16, const DateLUTImpl &)
+    {
+        throwDateIsNotSupported(name);
+    }
+    static constexpr bool hasPreimage() { return false; }
+
+    using FactorTransform = ToStartOfSecondImpl;
+};
+
```

Comment:
This implementation won't work well because you simply multiply `seconds` value by 1000. We also need to keep in mind that we can have milli- micro- nanosecond values in datetime. For example:
```SQL
SELECT toMillisecond(toDateTime64('2023-04-21 10:20:30.123456', 2));
```
will return `30 000`, but should return `30 123`

Reply:
I see, I misunderstood the intention of the function. I'll fix that.